### ISSUE TITLE
Links in Message Bar do not respond to High Contrast

### ISSUE THREAD

**Issue by betrue-final-final** [OfficeDev/office-ui-fabric-react#10880]
This reproes in all High Contrast themes
![image](https://user-images.githubusercontent.com/20363037/67025805-7f7d9d00-f0bb-11e9-97be-edef3223f9e6.png)

You can repro at https://aka.ms/fabricdemo
Change to any high contrast theme, and go to MessageBar. Turn on one of them with a link and observe.

The color should use link color for the theme.

### Priorities and help requested (not applicable if asking question):

Are you willing to submit a PR to fix? (Yes)

Requested priority: (High)

Products/sites affected: OneDrive, SharePoint, Outlook, various Azure properties

**Comment by ecraig12345** [comment]
Related: #10493

**Comment by ecraig12345** [comment]
@betrue-final-final What if the link color for the theme doesn't have high enough contrast with the particular MessageBar variant's background color? (Should we even be respecting the variant background in high contrast?)

**Comment by ecraig12345** [comment]
(@marygans, I'm probably going to take care of this one along with a couple other MessageBar/Link color issues.)

**Comment by khmakoto** [comment]
@ecraig12345 should we close either this or #10493 as duplicates? Seems like they are actually talking about the same thing. That way, we can track work on it in only one place.

**Comment by ecraig12345** [comment]
No, they're slightly different.

**Comment by ecraig12345** [comment]
Looking at this further I'm pretty sure it's a duplicate of #10493. Since that one has a solution outlined already, I'll close this one.

**Comment by ecraig12345** [comment]
Changed my mind, I do think now that this is a slightly different issue requiring design input. While using the default link color sounds good in theory, it's problematic in practice because of the MessageBar's inverted color scheme in high contrast.

As an example of the problem, here are all the default colors for high contrast black:
![image](https://user-images.githubusercontent.com/5864305/71124162-1b4d8700-2199-11ea-8c38-60c199a4a2a2.png)

Presumably to make the MessageBar stand out, we invert its colors and disable default automatic high contrast adjustments (which is normally how the high contrast link color is applied).
![image](https://user-images.githubusercontent.com/5864305/71124038-dde8f980-2198-11ea-87a3-5ef64287c9e9.png)

If you re-enable high contrast adjustment for links (`.ms-MessageBar .ms-Link:a { -ms-high-contrast-adjust: auto }`) they become unreadable:
![image](https://user-images.githubusercontent.com/5864305/71125031-070a8980-219b-11ea-81c6-11b4635a65f6.png)

So the question for @betrue-final-final is, since the default link color probably won't work on inverted backgrounds, what color do we use instead?

(Note that whatever change we make in the main styles will probably have to be duplicated in the website-specific Link overrides in Page.module.scss.)

**Comment by ecraig12345** [comment]
Note that the fix for this will have to take into account the selector conflict issue @khmakoto pointed out in https://github.com/OfficeDev/office-ui-fabric-react/issues/10493#issuecomment-540212625.

**Comment by betrue-final-final** [comment]
I’m suspecting we need to rethink the backgrounds here. I’ll come with a recommendation by the end of the week.

**Comment by ecraig12345** [comment]
Updates from Ben (thanks!)

![design](https://user-images.githubusercontent.com/5864305/72183421-d76a2e00-33a2-11ea-9880-dfe6603299a0.png)

All borders: Text
Text: Text color
Link: Link color
 
Info Fill - Background, Border
Warning fill - `rgba(255,254,57,0.3)`
Severe warning, Error, Blocked fill - `rgba(255,0,0,0.3)`
Success fill - `rgba(48,241,73,0.3)`

These colors should not be affected by user HC themes as the colors have meaning.

Ben confirmed the text colors work in HC black and white. Note that passing contrast requirement in HC is 7:1.
